tests/parser_tests.py

```diff
@@ -527,6 +530,25 @@ def test_parse_with_extra_words_at_start_and_end_valid(self):
             datetime(2016, 5, 16, 4, 5, 6, 789120),
         )
 
+        self.assertEqual(
+            self.parser.parse(
+                "Meet me at my house on the my birthday (2019-24-11)", "YYYY-DD-MM"
+            ),
+            datetime(2019, 11, 24),
+        )
+
+        self.assertEqual(
```

Comment:
I would add a comment to this stating that it fixes the case of a **partial** match that is surrounded by punctuation.